src/it/kotlin/GivenTreepatExpressionAndTreeThenFindMatchesLargeKtTest.kt

```diff
@@ -22,45 +25,26 @@ internal class GivenTreepatExpressionAndTreeThenFindMatchesLargeKtTest {
 
     private val resourcesLocation = "./src/it/resources/expressions_trees_matches/"
 
-    private fun parseTreeFile(tree: String): TargetTreeNode {
-        val lexer = TreeFormatLexer(CharStreams.fromString(tree))
-        val tokenStream = CommonTokenStream(lexer)
-        val fileParser = TreeFormatParser(tokenStream)
-        val treeFile: ParseTree = fileParser.subtree()
-        val treeVisitor = TreeFormatVisitorImplementation()
-        return treeVisitor.visit(treeFile)
-    }
-
-    private fun parseTreepat(tree: String): ASTNode {
-        val lexer = TreepatLexer(CharStreams.fromString(tree))
-        val tokenStream = CommonTokenStream(lexer)
-        val treepatParser = TreepatParser(tokenStream)
-        val treeTreepat: ParseTree = treepatParser.treepat()
-        val treepatVisitor = TreepatVisitorImplementation()
-        return treepatVisitor.visit(treeTreepat)
-    }
-
     private val runTest: GivenTreepatExpressionAndTreeThenFindMatchesFunction =
         { treepatAntlrString, treeAntlrString, output, error ->
             // Treepat Parsing
-            val astRoot: ASTNode = parseTreepat(treepatAntlrString)
-
+            val treepatExpression = TreepatExpression.createFromString(treepatAntlrString)
             // Tree File Parsing
-            val targetTreeNode: TargetTreeNode = parseTreeFile(treeAntlrString)
-
-            val rootFunctionModule = createVisitorFunction(astRoot)
+            val targetTree = DefaultTargetTree.createFromString(treeAntlrString)
+            // Match Manager
+            val manager = MatchManager(treepatExpression, targetTree)
+            val matches = manager.getAllMatches()
 
-            // act
-            val functionResult = rootFunctionModule.invoke(targetTreeNode)
+            val printableMatches: List<String>
 
-            val solutions: List<String> = if (functionResult.hasMatch) {
-                functionResult.responses.map { targetTreeNode.matchedNodesString(it.matches).trimIndent() }
+            if (matches.isNotEmpty()) {
+                printableMatches = matches.map{manager.matchToString(it)}
             } else {
-                listOf("Match not found")
+                printableMatches = listOf("Match not found")
             }
 
             // assert
-            assertEquals(output, solutions.joinToString("\n-\n"), error)
+            assertEquals(output, printableMatches.joinToString("\n-\n"), error)
```

Comment:
Also did this with the tests, so they would test all I did. This is an example of the use of `getAllMatches()`.